### DISCUSSION TITLE
How to get the Windows AUMID from the running installed application?

### DISCUSSION THREAD

**Discussion by eskatos** [hydraulic-software/conveyor#135] [Q&A] [answered]
The Windows variant of my JVM application needs to know the Windows AUMID at runtime.

I can get it using `conveyor make app-user-model-id` at buid time and then either hardcode it in my app or automate extracting it from conveyor's output during build time but this feels respectively clunky and overkill.

I figured the information I need is also stored in the site's `metadata.properties` under the  `windows.package-family-name` key. But I would like to have this information available in the app without doing any network request.

I looked at system properties and environment variables available at runtime but did not find anything.

Did I miss something that already exists that would provide this?
If not, could it be considered as a new feature request?

**Comment by eskatos** [comment]
Executing `Get-StartApps <app.name>` and parsing the output gives the right information. This looks like the best workaround so far.

But, given this information can be known at build time, running a separate process and parsing its output could be avoided.

**Comment by mikehearn** [comment, marked as answer]
The AUMID is stable for a given set of names and certificates. So, you don't necessarily have to run it every time, you can just run it once. As long as you don't change the subject name of your certificate, or the name of your app, it won't change.

We could definitely wire it up to a system property or environment variable however. I'll file an internal ticket for that.

**Reply by eskatos** [reply]
Thanks!

I think that wiring it to a system property or environment variable would be beneficial, especially during development time when names aren't set in stone yet. It would make for a better developer experience when starting with the product.

**Reply by mikehearn** [reply]
This is a good idea and will ship in the next release. On the JVM it'll be in the `app.windows.userModelID` system property, for Electron it'll be in the `appUserModelID` key in package.json

**Reply by mikehearn** [reply]
This has shipped! Thanks for the suggestion.
